[Visual Studio] Debug information was gone since 259957@main

basuke opened the pull request:
#### 954abe6a1af976007d7c62d14b9a1b3e61cdf267
<pre>
[Visual Studio] Debug information was gone since 259957@main
<a href="https://bugs.webkit.org/show_bug.cgi?id=252429">https://bugs.webkit.org/show_bug.cgi?id=252429</a>

Reviewed by Adrian Perez de Castro and Michael Catanzaro.

Disable fission settings when generator is Visual Studio because VS
doesn&apos;t handle debug information well if this settings is on for llvm.

* Source/cmake/OptionsCommon.cmake:

Canonical link: <a href="https://commits.webkit.org/260488@main">https://commits.webkit.org/260488@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e8f80d08bb38e6a373e7c3b67a96b8dcd0e369c5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/108379 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/17475 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/41244 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/117496 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/116864 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/18861 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/8755 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/100618 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/114146 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/14161 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/97412 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/42148 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/96138 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/29057 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/83837 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/97549 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/10307 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/30401 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/96888 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/8414 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/11050 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/7306 "Passed tests") | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/46/builds/30950 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/16457 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/49997 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/105966 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7253 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/12636 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/26228 "Passed tests") | 
| | | | | 
<!--EWS-Status-Bubble-End-->

aperezdc commented on Source/cmake/OptionsCommon.cmake:
```diff
@@ -153,20 +153,22 @@ if (USE_THIN_ARCHIVES)
-if (CXX_COMPILER_SUPPORTS_GSPLIT_DWARF AND LD_SUPPORTS_SPLIT_DEBUG)
-  set(ENABLE_DEBUG_FISSION_DEFAULT ON)
-endif ()
+if (NOT (CMAKE_GENERATOR MATCHES "Visual Studio"))
```
Wouldn't it be better to check `CMAKE_CXX_COMPILER_ID NOT STREQUAL MSVC`? I suppose nobody would use the Visual Studio generator with a different compiler than MSVC, but one can reasonable use Make/Ninja with MSVC (AFAIU).

basuke replied:
Unfortunately no. PlayStation always uses LLVM compiler while using Visual Studio generator for debugging. That's why this default becomes on. And that breaks VS debug info.

aperezdc replied:
Thanks for the clarification, that helps 👌🏼

philn reviewed (commented):
LGTM but i'll let real CMake experts sign-off on this...

mcatanzaro reviewed (approved):
It might be nicer to write this as:

```
set(ENABLE_DEBUG_FISSION_DEFAULT OFF)
check_cxx_compiler_flag(-gsplit-dwarf CXX_COMPILER_SUPPORTS_GSPLIT_DWARF)
if (CXX_COMPILER_SUPPORTS_GSPLIT_DWARF AND LD_SUPPORTS_SPLIT_DEBUG AND NOT CMAKE_GENERATOR MATCHES "Visual Studio")
  set(ENABLE_DEBUG_FISSION_DEFAULT ON)
endif ()
```

Or if you want to avoid the extra compiler flag test:

```
set(ENABLE_DEBUG_FISSION_DEFAULT OFF)
if (NOT CMAKE_GENERATOR MATCHES "Visual Studio")
    check_cxx_compiler_flag(-gsplit-dwarf CXX_COMPILER_SUPPORTS_GSPLIT_DWARF)
    if (CXX_COMPILER_SUPPORTS_GSPLIT_DWARF AND LD_SUPPORTS_SPLIT_DEBUG AND NOT CMAKE_GENERATOR MATCHES "Visual Studio")
      set(ENABLE_DEBUG_FISSION_DEFAULT ON)
    endif ()
endif ()
```

The sccache stuff will go away eventually and so is best kept in its own separate messy section, but this part will stay and I don't see the need for it to be separate at the bottom.

basuke committed:
[Visual Studio] Debug information was gone since 259957@main
https://bugs.webkit.org/show_bug.cgi?id=252429

Reviewed by Adrian Perez de Castro and Michael Catanzaro.

Disable fission settings when generator is Visual Studio because VS
doesn't handle debug information well if this settings is on for llvm.

* Source/cmake/OptionsCommon.cmake:

Canonical link: https://commits.webkit.org/260488@main
